services/offline/src/lib/cacheable-section.js

```diff
@@ -0,0 +1,176 @@
+import PropTypes from 'prop-types'
```

Comment:
This file includes context for cacheable sections' recording states, the `useCacheableSection` hook, and the `CacheableSection` component wrapper.